libraries/lib-numeric-formats/formatters/ParsedNumericConverterFormatter.cpp

```diff
@@ -580,6 +581,403 @@ class ParsedNumericConverterFormatter final : public NumericConverterFormatter
+         * to '>' if your language uses a '.'. */
+         XO("01000>01000 kHz|0.001")
+         , XO("hertz")
+      }
+   },
+};
+
+/** \brief array of formats the control knows about internally
+ *  array of string pairs for name of the format and the format string
+ *  needed to create that format output. This is used for the pop-up
+ *  list of formats to choose from in the control. */
+static const BuiltinFormatString BandwidthConverterFormats_[] = {
+   {
+   /* i18n-hint: Name of display format that shows log of frequency
+    * in octaves */
+   { XO("octaves") },
+   {
+      /* i18n-hint: Format string for displaying log of frequency in octaves.
+       * Change the decimal points for your locale. Don't change the numbers.
+       * The decimal separator is specified using '<' if your language uses a ',' or
+       * to '>' if your language uses a '.'. */
+      XO("100>01000 octaves|1.442695041"),    // Scale factor is 1 / ln (2)
+      /* i18n-hint: an octave is a doubling of frequency */
+      XO("thousandths of octaves")
+   }
+   },
+
+   {
+   /* i18n-hint: Name of display format that shows log of frequency
+    * in semitones and cents */
+   { XO("semitones + cents") },
+   {
+      /* i18n-hint: Format string for displaying log of frequency in semitones
+       * and cents.
+       * Change the decimal points for your locale. Don't change the numbers.
+       * The decimal separator is specified using '<' if your language uses a ',' or
+       * to '>' if your language uses a '.'. */
+      XO("1000 semitones >0100 cents|17.312340491"),   // Scale factor is 12 / ln (2)
+      /* i18n-hint: a cent is a hundredth of a semitone (which is 1/12 octave) */
+      XO("hundredths of cents")
+   }
+   },
+   
+   {
+   /* i18n-hint: Name of display format that shows log of frequency
+    * in decades */
+   { XO("decades") },
+   {
+      /* i18n-hint: Format string for displaying log of frequency in decades.
+       * Change the decimal points for your locale. Don't change the numbers. */
+      XO("10>01000 decades|0.434294482"),   // Scale factor is 1 / ln (10)
+      /* i18n-hint: a decade is a tenfold increase of frequency */
+      XO("thousandths of decades")
+   }
+   },
+};
+
+Registry::BaseItemPtr MakeGroup (
+   const Identifier& identifier, NumericConverterType type, const BuiltinFormatString* formatStrings, const size_t count)
+{
+   auto group =
+      std::make_unique<NumericConverterRegistryGroup>(identifier, type);
+
+   for (size_t index = 0; index < count; ++index)
+   {
+      const auto functionIdentfier =
+         wxString::Format(L"%s_%zu", identifier.GET(), index);
+      
+      auto& formatString = formatStrings[index];
+
+      group->items.push_back(std::make_unique<NumericConverterRegistryItem>(
+         functionIdentfier, formatString.name,
+         formatString.formatStrings.fraction,
+         [&formatString, type](const auto& cfg)
```

Comment:
That looks like a dangerous by-reference capture, but it isn't so long as the tables remain static... worth a commen

Reply:
This was not intentional